lib/resources/nginx.rb

```diff
@@ -0,0 +1,108 @@
+        its('group') { should be 'root' }
+      end
+    "
+
+    def initialize
+      return skip_resource 'The `nginx` resource is not yet tested on your OS.' if inspec.os.windows?
+
+      cmd = inspec.command('nginx -V 2>&1')
+      if !cmd.exit_status.zero?
+        return skip_resource 'Error using the command nginx -V'
+      end
+      @data = cmd.stdout
+      @params = {}
+      read_content
+    end
+
+    # @todo RFC: which of these methods should be private and or not exposed?
+    #
+
+    def method_missing(name)
+      # ensure params are loaded
+      read_content
+      # extract values
+      @params[name.to_s] unless @params.nil?
+    end
+
+    def read_content
+      parse_config
+      parse_path
+      parse_http_path
+    end
+
+    def parse_config
+      @params['prefix'] = @data.scan(/--prefix=(\S+)\s/).flatten.first
+      @params['service'] = 'nginx'
+      @params['user'] = @data.scan(/--user=(\S+)\s/).flatten.first
+      @params['group'] = @data.scan(/--group=(\S+)\s/).flatten.first
+      @params['version'] = @data.scan(%r{nginx version: nginx\/(\S+)\s}).flatten.first
+    end
+
+    def parse_path
+      @params['conf_path'] = @data.scan(/--conf-path=(\S+)\s/).flatten.first
+      @params['conf_dir'] = Pathname.new(@params['conf_path']).dirname.to_s
+      @params['sbin_path'] = @data.scan(/--sbin-path=(\S+)\s/).flatten.first
+      @params['modules_path'] = @data.scan(/--modules-path=(\S+)\s/).flatten.first
+      @params['error_log_path'] = @data.scan(/--error-log-path=(\S+)\s/).flatten.first
+      @params['http_log_path'] = @data.scan(/--http-log-path=(\S+)\s/).flatten.first
+      @params['pid_path'] = @data.scan(/--pid-path=(\S+)\s/).flatten.first
+      @params['lock_path'] = @data.scan(/--lock-path=(\S+)\s/).flatten.first
+    end
+
+    def parse_http_path
+      @params['http_client_body_temp_path'] = @data.scan(/--http-client-body-temp-path=(\S+)\s/).flatten.first
+      @params['http_proxy_temp_path'] = @data.scan(/--http-proxy-temp-path=(\S+)\s/).flatten.first
+      @params['http_fastcgi_temp_path'] = @data.scan(/--http-fastcgi-temp-path=(\S+)\s/).flatten.first
+      @params['http_uwsgi_temp_path'] = @data.scan(/--http-uwsgi-temp-path=(\S+)\s/).flatten.first
+      @params['http_scgi_temp_path'] = @data.scan(/--http-scgi-temp-path=(\S+)\s/).flatten.first
+    end
+
+    def openssl_version
+      result = @data.scan(/built with OpenSSL\s(\S+)\s(\d\d\s\S+\s\d\d\d\d)/).flatten
+      openssl_version = { 'version' => result[0], 'date' => result[1] }
+      openssl_version unless openssl_version.nil?
+    end
+
+    def compiler_info
+      result = @data.scan(/built by (\S+)\s(\S+)\s(\S+)/).flatten
+      compiler_info = { 'compiler' => result[0], 'version' => result[1], 'date' => result[2] }
+      compiler_info unless compiler_info.nil?
+    end
+
+    def support_info
+      support_info = @data.scan(/(.*\S+) support enabled/).flatten.join(' ')
+      support_info unless support_info.nil?
```

Comment:
This will never be nil, and you want to return nil if it's not available.

I'd write this to be:

```ruby
support_info = @data.scan(/(.*\S+) support enabled/).flatten
support_info.empty? ? nil : support_info.join(' ')
```